### PR TITLE
Fix client bundle issue

### PR DESCRIPTION
**Pull request by thomasdigby**
#### What have you done
Fixed an issue with our client `webpack` setup where `tapestry-wp` was not getting transpiled

#### Why have you done it
Because we need to transpile it

#### Testing carried out to prevent breaking changes
No broken tests

### CODE REVIEW

**Comment by thomasdigby** [comment]
This works great for development mode, but I'm running into a whole bunch of problems on production build. Hold merging this for now.

File: bin/tapestry-build [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env node
+
+'use strict'
```
Comment (mathewhawley) [inline bin/tapestry-build]: Small thing – we probably don't need this if we're running in `>= v6.9`?

File: bin/tapestry-build [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    })
+  })
+
+glob
+  .use(noTests)
+  .use(noPublic)
+  .use(noScripts)
+  .readdir('**/*.js', (err, files) => {
+    if (err) return logger.error(err)
+    files.map(file => transform(file))
```
Comment (mathewhawley) [inline bin/tapestry-build]: Might be going blind – but the array return from this `.map` isn't being used anywhere? Would `.forEach()` work better instead?

File: src/index.js [hunk 3]
```diff
@@ -12,7 +14,12 @@ export const client = (options) => new Client(options)
 // create client bundle and boot server on callback, avoids the server booting without the client ready. Object.assign() used instead of spread operator as we're only supporting es2015 (currently)
 export const boot = (options) => new Client(
   Object.assign({
-    onComplete: () => new Server(options)
+    onComplete: () => {
+      // get bundle data and pass through to server
+      const assetsPath = path.resolve(options.cwd, '.tapestry', 'assets.json')
+      const assets = fs.readJsonSync(assetsPath)
+      return new Server(Object.assign(options, { assets }))
```
Comment (mathewhawley) [inline src/index.js]: Would it be safer to pass an empty `target` to `Object.assign()`? e.g. `Object.assign({}, ...)` – unless mutation of `options` is intentional?

File: bin/tapestry-build (hunk 2 above)
Comment (thomasdigby) [inline bin/tapestry-build]: It would!

File: bin/tapestry-build (hunk 1 above)
Comment (thomasdigby) [inline bin/tapestry-build]: Probably not yes

File: bin/tapestry-build (hunk 2 above)
Comment (jonsherrard) [inline bin/tapestry-build]: Using map when there's no array returned is just a stylistic thing isn't. I prefer map as we use it elsewhere. Also mutating the original files array feels... like there could be side effects? I like that map creates a copy and iterates over that. Less prone to error.

Comment (jonsherrard) [inline bin/tapestry-build]: It also allows us to add to the chain later on without refactoring from forEach, and I believe it's faster as an iterator than forEach.

File: src/server/render/default-html.js [hunk 4]
```diff
@@ -6,19 +6,13 @@ const defaultHtml = ({ markup, head, asyncProps, assets = {} }) => {
...
-        {
-          assets.bundle &&
-            <script defer src={assets.bundle.js} />
-        }
+        { head.title.toComponent() }
+        { head.base.toComponent() }
+        { head.meta.toComponent() }
+        { head.link.toComponent() }
+        { head.script.toComponent() }
+        { assets.vendor && <script defer src={assets.vendor.js} /> }
```
Comment (jonsherrard) [inline src/server/render/default-html.js]: Lovely.

File: bin/tapestry-build (hunk 2 above)
Comment (mathewhawley) [inline bin/tapestry-build]: Ok sure, it's a little confusing to read right now, because you'd expect the returned array to be used for something?

**Pull request merged**